packages/itwinui-css/src/utils/notification-marker/notification-marker.scss

```diff
@@ -2,25 +2,38 @@
 // See LICENSE.md in the project root for license terms and full copyright notice.
 
 $iui-notification-marker-border-width: var(--iui-size-3xs);
+$iui-notification-before-size: calc(var(--iui-size-xs) + $iui-notification-marker-border-width * 2);
+$iui-notification-before-size-radius: calc($iui-notification-before-size * 0.5);
 
 @mixin iui-notification-marker($status: primary) {
   position: relative;
 
   &::before {
     content: '';
     position: absolute;
-    width: calc(var(--iui-size-xs) + $iui-notification-marker-border-width * 2);
-    height: calc(var(--iui-size-xs) + $iui-notification-marker-border-width * 2);
-    top: calc(var(--iui-size-s) * -0.5);
-    right: calc(var(--iui-size-s) * -0.5);
+    width: calc($iui-notification-before-size - 2 * $iui-notification-marker-border-width);
+    height: calc($iui-notification-before-size - 2 * $iui-notification-marker-border-width);
+    top: calc($iui-notification-before-size-radius + var(--iui-size-s) * -0.5);
+    right: calc($iui-notification-before-size-radius + var(--iui-size-s) * -0.5);
+    transform: translate(50%, -50%);
+
     border-radius: 100%;
     background-color: var(--iui-color-foreground-#{$status});
-    border: $iui-notification-marker-border-width solid var(--iui-color-background-1);
     @media (prefers-reduced-motion: no-preference) {
       transition: background-color var(--iui-duration-1) ease-out, border-color var(--iui-duration-1) ease-out;
     }
   }
 
+  > * {
```

Comment:
Since we now rely on always having the svg as a child, I would suggest updating the html so that the notification-marker class is applied on a wrapper element (e.g. `span`) rather than the custom `svg-` element

Reply:
**[DONE]** I changed the dom structure for all the `.iui-button-icon`s with the notification marker (without notifications icons are unchanged).

Old
![image](https://user-images.githubusercontent.com/45748283/193142686-6c5d655a-e314-4808-a688-a5612594bdbf.png)

New
![Screenshot from 2022-09-29 13-37-22](https://user-images.githubusercontent.com/45748283/193142422-592919fb-d8f5-4250-8fba-345fc8b6fcb3.png)

Reply:
the `iui-button-icon` class shouldn't go on the span, it should go on the svg

Reply:
I did this to match what happens in React. `IconButton` adds the `.iui-button-icon` to the child tag itself (Using `React.cloneChild`. [Code](https://github.com/iTwin/iTwinUI-react/blob/main/packages/iTwinUI-react/src/core/Buttons/IconButton/IconButton.tsx#L61-L67)). 

As a result, the below React code gives the below dom:

![image](https://user-images.githubusercontent.com/45748283/193333636-1280f9ed-b147-49ad-9cd0-a833d4704590.png)

![image](https://user-images.githubusercontent.com/45748283/193333406-f4f80381-eb35-42f1-8e8d-cc725c1aeafd.png)

In the above screenshot, `.iui-button-icon` is on the `span`, which is why I did the same in CSS. Also, I couldn't get it to work in CSS by putting `.iui-button-icon` on the svg and not the span, but I can try again.

Reply:
> I did this to match what happens in React. `IconButton` adds the `.iui-button-icon` to the child tag itself (Using `React.cloneChild`. [Code](https://github.com/iTwin/iTwinUI-react/blob/main/packages/iTwinUI-react/src/core/Buttons/IconButton/IconButton.tsx?rgh-link-date=2022-09-30T18%3A28%3A19Z#L61-L67)).

Then we might need to change the approach, instead of working around it in css.

> I couldn't get it to work in CSS by putting `.iui-button-icon` on the svg and not the span, but I can try again.

What problem where you having?

I would much rather _always_ have a span for `iui-button-icon` or _never_ have it, rather than trying to cover both cases like we need to do now.

Reply:
> What problem where you having?

After adding `.iui-button-icon` to the svg-document, the width of it goes to 0 (`0x24`). Even adding a `width: 24px` inline style doesn't fix it. I could look into it more, but here's what I noticed so far.

![image](https://user-images.githubusercontent.com/45748283/194155283-25f04c38-5c27-42a3-beeb-9dd3424602f9.png)

Reply:
> the iui-button-icon class shouldn't go on the span, it should go on the svg

I changed it to apply `.iui-button-icon` on the `svg-document` and not the wrapper span.

##### Previous
![Screenshot from 2022-09-29 13-37-22](https://user-images.githubusercontent.com/45748283/193142422-592919fb-d8f5-4250-8fba-345fc8b6fcb3.png)

##### Latest
![image](https://user-images.githubusercontent.com/45748283/194917816-4c7dda77-6d61-4a17-a6f6-4a2240225814.png)

1. The `0` width problem was because of `overflow: hidden` property near [L107](https://github.com/iTwin/iTwinUI/pull/768/files/8398b1e7a2580b66e959a441765316a28fa2bbec..acd42e3e527d13ef1ee2fe19ca667b53955ce87a#diff-187108cb5bf3f6ce0ad8a514769689db86ffe4102ee775050d0b4c02dda54bedR107). I changed that to only target the label span and not the motivation marker span.

2. There was also a `margin-block` (orange portion in screenshot) on the `.iui-button-icon` that was causing the circle's incorrect vertical placement.
![image](https://user-images.githubusercontent.com/45748283/194916411-b62d6ed5-24b1-4a4a-9b24-87453d5e61e1.png)
![image](https://user-images.githubusercontent.com/45748283/194916394-0753e6ee-7e3c-466a-9b41-2694fbd62b46.png)
I fixed this by removing this when the root of the `.iui-button-icon` has `.iui-notification-marker` [L41-44](https://github.com/iTwin/iTwinUI/pull/768/files/8398b1e7a2580b66e959a441765316a28fa2bbec..acd42e3e527d13ef1ee2fe19ca667b53955ce87a#diff-b2e4ed571b9f44ba530e2dfa36a85a31706270f62ca9fc97791963f55ce101feR41-R44)

All tests still pass, but just to confirm, would the changes in `button.scss` and `side-navigation.scss` in these [two commits](https://github.com/iTwin/iTwinUI/pull/768/files/8398b1e7a2580b66e959a441765316a28fa2bbec..acd42e3e527d13ef1ee2fe19ca667b53955ce87a#diff-b2e4ed571b9f44ba530e2dfa36a85a31706270f62ca9fc97791963f55ce101fe) break anything?

Also, for the React side, I guess we have to think of how to implement it there, since there, `.iui-button-icon` is applied on the wrapper `span`

Reply:
I think we might want to do away with that `margin-block`. It was added to make the height of the small icon button match the height of the small text button.

We explicitly set the `min-height` to 24px, but since the line-height of text is already 24px and there's a 1px on each side, the total comes out to be 26px. To match this 24px text, we add 4px margin above and below the 16px icon.

We could fix this one of two ways (@FlyersPh9 might want to comment on this part):
1. Set min-height to explicitly be 26px instead of 24px, OR
2. Decrease line-height so that small button actually becomes 24px tall instead of 26px.

---

As for React, I think our users are already having issues with `React.cloneElement`, so we might want to change the approach to _always_ have a wrapping span. The DOM structure could look something like this:
```html
<button>
  <span class='iui-button-icon'>
    <svg>...</svg>
  </span>
  <span>Text</span>
</button>
```

And then react could would just wrap the icon instead of cloning (similar change in IconButton):

```jsx
const Button = ({ startIcon, children }) => (
  <button>
    <span class='iui-button-icon'>{startIcon}</span>
    <span>{children}</span>
  </button>
);
```

Based on that react change, if the user were to do this:

```jsx
<IconButton>
  <NotificationMarker>
    <SvgSomething />
  </NotificationMarker>
</IconButton>
```

it would result in this markup:
```html
<button>
  <span class='iui-button-icon'>
    <span class='iui-notification-marker'>
      <svg>...</svg>
    </span>
  </span>
</button>
```

which shouldn't be a problem for us if we make sure our `iui-button-icon` handles svgs nested deeply like this:

```css
.iui-button-icon {
  svg {
    width: 16px;
    height: 16px;
    fill: currentColor;
  }
}
```

Reply:
I just now tried removing `line-height` from the button and `margin-block` from the button icon.  Height wise, I like that the small button is now 24px tall.  What I don't like is that the text label appears to be slightly off center vertically.  Maybe playing around with the `line-height` will resolve this?
![Screen Shot 2022-10-12 at 10 28 21 AM](https://user-images.githubusercontent.com/849817/195370260-50335c19-2e57-4a5c-90e0-53439cdc3b71.png)

Reply:
`line-height: 1.2` seems to work

Reply:
I made the following changes yesterday:
* In the CSS repo, I changed the line height to 1 ([Commit](https://github.com/iTwin/iTwinUI/pull/768/commits/e015a91490bd6fdf629261ae96092ee2b180998c)). I thought of line height of 1.2, as Mayank suggested, but saw now difference between 1 and 1.2. Using 1 also solved the alignment problem Jon was mentioning above. Both red rectangles are 20px tall.
  * ![button-line-height](https://user-images.githubusercontent.com/45748283/195906958-22ddc342-6415-4eef-b672-b4e51bfa4dcd.png)
  * Also the small icon button and small text button are 24px tall.
  * This led to 69 test images fail, because buttons are used in many components.

* In the React repo's NotificationMarker PR, I removed `React.cloneElement` ([Commit](https://github.com/iTwin/iTwinUI-react/pull/829/commits/48ae082db32515d384fa2f79cae47ab634435888)).
  * 4 unit tests failed because of incorrect selectors after this change (e.g. using `>` selector or `.firstElementChild`).

Reply:
i saw a difference between `1` and `1.2` on my machine (better aligned with `1.2`) and also if the button ever has multiple lines of text for some reason, 1.2 will look better

Reply:
Ohh, I think the multiple lines point is important, thanks. Changed `line-height` to `1.2`
`line-height: 1`
![image](https://user-images.githubusercontent.com/45748283/196263819-c6174937-ad6a-4368-976d-3d8b1c90123a.png)

`line-height: 1.2`
![image](https://user-images.githubusercontent.com/45748283/196263886-e6d72a9c-1c16-49ea-b4de-b1c5584a7c2d.png)

Reply:
Is it possible that browser rendering differences could cause unexpected results? I tested `line-height` with `1` and `1.2` on the `buttons.html -> Default -> Small Right Icon` manually/visually on Windows and Linux and got different results.

1. On Windows, `1` and `1.2` gave **same results**. The top and bottom gaps were 8 and 9 units respectively. Both red rectangles are 8 units tall.

##### 1 and 1.2
![image](https://user-images.githubusercontent.com/45748283/197068315-165cf307-872c-4ed0-89d4-bba702ecc9ea.png)

2. On Linux, `1` and `1.2` gave slightly **different results**. the top and bottom gaps were 9 & 12 units for `1`, and 10 & 11 units for `1.2`. The red rectangles are 9 units tall for `1` and 10 units tall for `1.2`

##### 1
![MicrosoftTeams-image (5)](https://user-images.githubusercontent.com/45748283/197070883-89a91af9-b824-4f9f-bbfd-0e5e7d0b8aa2.png)

##### 1.2
![MicrosoftTeams-image (4)](https://user-images.githubusercontent.com/45748283/197070846-9a47e731-d6f9-444a-b595-1d5f2c49bc0c.png)

3. When running automated tests (`yarn test`), with `1`, **75 tests failed** and with `1.2`, **95 tests failed**.

Since the automated tests always run in the same environment, and since `1` caused the least test changes, I chose the `line-height` to be `1`. What do you think @mayank99 @FlyersPh9?

Reply:
As per our previous discussion, changed the `line-height: 1.2` since that seems to be the most vertically centered in the screenshots and on Mayank's system.

I made these non-notification related changes in CSS PR [#788](https://github.com/iTwin/iTwinUI/pull/788) and React PR [#895](https://github.com/iTwin/iTwinUI-react/pull/895).